WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -0,0 +1,207 @@
+        /// A theme's CSS will take into account this number to decide whether to responsively display its layout to be
+        /// the mobile, tablet, or desktop layout.
+        ///
+        var browserWidth: CGFloat {
+            switch self {
+            case .mobile: return 400
+            case .tablet: return 800
+            case .desktop: return 1200
+            }
+        }
+
+        /// This JavaScript forces the WebView's viewport to match the supplied width above. This allows the preview
+        /// functionality to switch between a theme's mobile, tablet, or desktop layout.
+        ///
+        var viewportScript: String {
+            let js = """
+            // remove all existing viewport meta tags - some themes included multiple, which is invalid
+            document.querySelectorAll("meta[name=viewport]").forEach( e => e.remove() );
+            // create our new meta element
+            const viewportMeta = document.createElement("meta");
+            viewportMeta.name = "viewport";
+            viewportMeta.content = "width=%1$d";
+            // insert the correct viewport meta tag
+            document.getElementsByTagName("head")[0].append(viewportMeta);
+            """
+
+            return String(format: js, NSInteger(browserWidth))
+        }
+    }
+
+    @State private var selectedDevice: PreviewDevice = PreviewDevice.defaultDevice
+    private let theme: WordPressTheme
+
+    /// Triggered when "Start with this theme" button is tapped.
+    var onStart: () -> Void
+
+    init(theme: WordPressTheme, onStart: @escaping () -> Void) {
+        self.theme = theme
+        self.onStart = onStart
+    }
+
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 0) {
+                if let url = theme.themeThumbnailURL {
+                    WebView(
+                        isPresented: .constant(true),
+                        url: url,
+                        shouldReloadOnUpdate: true,
+                        onCommit: { webView in
+                            webView.evaluateJavaScript(self.selectedDevice.viewportScript)
+                        }
+                    )
+
+                    Divider()
+                        .frame(height: Layout.dividerHeight)
+                        .foregroundColor(Color(.divider))
+
+                    VStack {
+                        Button(Localization.startWithThemeButton, action: onStart)
+                        .buttonStyle(PrimaryButtonStyle())
+
+                        Text(String(format: Localization.themeName, theme.name))
+                            .secondaryBodyStyle()
+                    }.padding(Layout.footerPadding)
+
+                } else {
+                    errorView
+                }
+            }
+            .toolbar {
+                if let url = theme.themeThumbnailURL {
+                }
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Image(uiImage: .closeButton)
```

Comment:
Should we use a button and add a callback closure for dismissing the view?

Reply:
yup! https://github.com/woocommerce/woocommerce-ios/pull/11452/commits/fc82c2a1a16754b2008591cd2a587f795757966d